index.js

```diff
@@ -2,33 +2,35 @@
 
 /* eslint-disable no-underscore-dangle */
 class NotificationBase {
-
     /**
      * Constructor for Build Notification
      * @method constructor
      * @param {Object}      config              Configuration specific to the plugin
-     * @param {Object}      server              API server
-     * @param {String}      eventName           Event name the notification plugin listen to
      * @return {NotificationBase}
      */
-    constructor(config, server, eventName) {
+    constructor(config) {
         this.config = config;
-        this.server = server;
-        this.eventName = eventName;
+    }
+
+    /**
+     * Get event list for receiving Screwdriver API events
+     * @method events
+     * @return {Array}
+     */
+    get events() {
+        return ['build_status'];
     }
 
     /**
      * Listen to the event and notify users based on their job settings
      * @method notify
-     * @return {Promise}
      */
-    notify() {
-        return this._notify();
+    notify(buildData) {
+        return this._notify(buildData);
```

Comment:
Each notification plugin should do try-catch in it. Catching all errors here is a last resort to avoid hapi event broken.